src/core/thread/peer.hpp

```diff
@@ -86,8 +95,25 @@ class Peer : public CslNeighbor
      */
     const Mle::TxChallenge &GetChallenge(void) const { return mAttachChallenge; }
 
+    /**
+     * Increments the count of re-transmitted link tear down messages.
+     */
+    void IncrementTearDownCount(void) { mTearDownCount++; }
+
+    /**
+     * Resets the count of re-transmitted link tear down messages to zero.
+     */
+    void ResetTearDownCount(void) { mTearDownCount = 0; }
+
+    /**
+     * Returns the count of re-transmitted link tear down messages.
+     */
+    uint8_t GetTearDownCount(void) const { return mTearDownCount; }
+
 private:
     Mle::TxChallenge mAttachChallenge;
+
+    uint8_t mTearDownCount : 3; ///< The count of re-transmitted link tear down messages.
```

Comment:
Since it is private, can remove `///<` (to avoid being added into doxygen?)
```suggestion
    uint8_t mTearDownCount : 3; // The count of re-transmitted link tear down messages.
```